src/main/java/tars/Storage.java

```diff
@@ -0,0 +1,144 @@
+                String description = parts[2];
+
+                try {
+                    switch (type) {
+                    case "T":
+                        tasks.add(new Todo(description, isDone));
+                        break;
+                    case "D":
+                        if (parts.length < 4) {
+                            throw new TarsException("Corrupt deadline task data: " + line);
+                        }
+                        String by = parts[3];
+                        tasks.add(new Deadline(description, isDone, by));
+                        break;
+                    case "E":
+                        if (parts.length < 5) {
+                            throw new TarsException("Corrupt event task data: " + line);
+                        }
+                        String from = parts[3];
+                        String to = parts[4];
+                        tasks.add(new Event(description, isDone, from, to));
+                        break;
+                    default:
+                        throw new TarsException("Unknown task type: " + type);
+                    }
+                } catch (DateTimeParseException e) {
+                    throw new TarsException("Invalid date format in task data: " + line);
+                }
+            }
+        } catch (IOException e) {
+            throw new TarsException("Error reading from file: " + e.getMessage());
+        }
+        return tasks;
+    }
+
+    /**
+     * Saves the given list of tasks to the file specified by the file path.
+     *
+     * <p>Each task is converted to a string in a specific format and written
+     * to the file. If the file's directory does not exist, it will be created.
+     *
+     * @param tasks the list of tasks to save to the file.
+     * @throws TarsException if there is an error writing to the file.
+     */
+    public void saveTasks(List<Task> tasks) throws TarsException {
+        File file = new File(filePath);
+        File directory = file.getParentFile();
+        if (!directory.exists()) {
+            try {
+                directory.mkdirs(); // Create directory
+            } catch (SecurityException e) {
+                System.out.println("There was a security/permission issue: " + e.getMessage());
+            }
+        }
+
+        try (BufferedWriter bw = new BufferedWriter(new FileWriter(file))) {
+            for (Task task : tasks) {
+                bw.write(taskToFileString(task));
+                bw.newLine();
+            }
+        } catch (IOException e) {
+            throw new TarsException("Error writing to file: " + e.getMessage());
+        }
+    }
+
+    /**
+     * Converts a Task object into a formatted string for saving to a file.
+     *
+     * @param task the task to convert to a string.
+     * @return a string representation of the task in the save file format.
+     */
+    private String taskToFileString(Task task) {
```

Comment:
Maybe you can make use of polymorphism and have a 'getType' method in the different Task child classes and let dynamic binding resolve this for you